third_party/move/move-vm/runtime/src/runtime.rs

```diff
@@ -1,310 +0,0 @@
-// Copyright (c) The Diem Core Contributors
-// Copyright (c) The Move Contributors
-// SPDX-License-Identifier: Apache-2.0
-
-use crate::{
-    data_cache::TransactionDataCache,
-    interpreter::Interpreter,
-    loader::LoadedFunction,
-    module_traversal::TraversalContext,
-    native_extensions::NativeContextExtensions,
-    session::SerializedReturnValues,
-    storage::{code_storage::CodeStorage, module_storage::ModuleStorage},
-    AsFunctionValueExtension, LayoutConverter, StorageLayoutConverter,
-};
-use move_binary_format::{
-    errors::{Location, PartialVMError, PartialVMResult, VMResult},
-    file_format::LocalIndex,
-};
-use move_core_types::{language_storage::TypeTag, value::MoveTypeLayout, vm_status::StatusCode};
-use move_vm_metrics::{Timer, VM_TIMER};
-use move_vm_types::{
-    gas::GasMeter,
-    loaded_data::runtime_types::Type,
-    value_serde::ValueSerDeContext,
-    values::{Locals, Reference, VMValueCast, Value},
-};
-use std::borrow::Borrow;
-
-/// An instantiation of the MoveVM.
-pub(crate) struct VMRuntime {}
-
-impl VMRuntime {
-    /// Creates a new runtime instance with provided environment.
-    #[allow(clippy::new_without_default)]
-    pub(crate) fn new() -> Self {
-        Self {}
-    }
-
-    fn deserialize_arg(
-        &self,
```

Comment:
wow this doesn't warn that self is unused?

Reply:
true, I'm used to having those as free functions, but it can make sense indeed, it's not the same as any other unused param